src/System.Buffers.Experimental/System/Buffers/SpanWriter.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Text;
+
+namespace System.Buffers
+{
+    public ref struct SpanWriter
+    {
+        Span<byte> _buffer;
+        int _written;
+        SymbolTable _symbols;
+
+        public ReadOnlyMemory<byte> Newline;
+        public Func<int, Memory<byte>> Enlarge;
+
+        static ReadOnlyMemory<byte> s_defaultNewline = new byte[] { (byte)'\n' };
+
+        public SpanWriter(Span<byte> buffer, SymbolTable symbols = default(SymbolTable))
+        {
+            _buffer = buffer;
+            _written = 0;
+            _symbols = symbols == default(SymbolTable) ? SymbolTable.InvariantUtf8 : symbols;
+            Newline = s_defaultNewline;
+            Enlarge = null;
+        }
+
+        public void WriteBytes<T>(T value, IBufferTransformation transformation = null) where T : IWritable
+        {
+            var start = Index;
```

Comment:
Do we need `start`? Why not just use `_written` to slice the buffer?

Reply:
I think you are right. I will fix